tests/components/vacuum/test_init.py

```diff
@@ -1,9 +1,144 @@
 """The tests for the Vacuum entity integration."""
 from __future__ import annotations
 
-from homeassistant.components.vacuum import StateVacuumEntity, VacuumEntityFeature
+from typing import Any
+
+import pytest
+
+from homeassistant.components.vacuum import (
+    DOMAIN,
+    SERVICE_CLEAN_SPOT,
+    SERVICE_LOCATE,
+    SERVICE_PAUSE,
+    SERVICE_RETURN_TO_BASE,
+    SERVICE_SEND_COMMAND,
+    SERVICE_SET_FAN_SPEED,
+    SERVICE_START,
+    SERVICE_STOP,
+    STATE_CLEANING,
+    STATE_IDLE,
+    STATE_PAUSED,
+    STATE_RETURNING,
+    StateVacuumEntity,
+    VacuumEntityFeature,
+)
 from homeassistant.core import HomeAssistant
 
+from tests.components.vacuum import create_entity
+from tests.testing_config.custom_components.test import vacuum as VacuumPlatform
+
+
+@pytest.mark.parametrize(
+    ("service", "expected_state"),
+    [
+        (SERVICE_CLEAN_SPOT, STATE_CLEANING),
+        (SERVICE_PAUSE, STATE_PAUSED),
+        (SERVICE_RETURN_TO_BASE, STATE_RETURNING),
+        (SERVICE_START, STATE_CLEANING),
+        (SERVICE_STOP, STATE_IDLE),
+    ],
+)
+async def test_state_services(
+    hass: HomeAssistant, config_flow_fixture: None, service: str, expected_state: str
+) -> None:
+    """Test get vacuum service that affect state."""
+
+    entity0 = await create_entity(hass, None)
+
+    _ = await hass.services.async_call(
```

Comment:
```suggestion
    await hass.services.async_call(
```
And all the places below